react/features/chat/functions.js

```diff
@@ -7,39 +7,56 @@ import { escapeRegexp } from '../base/util';
 
 /**
  * An ASCII emoticon regexp array to find and replace old-style ASCII
- * emoticons (such as :O) to new Unicode representation, so then devices
- * and browsers that support them can render these natively without
- * a 3rd party component.
+ * emoticons (such as :O) with the new Unicode representation, so that
+ * devices and browsers that support them can render these natively
+ * without a 3rd party component.
  *
  * NOTE: this is currently only used on mobile, but it can be used
  * on web too once we drop support for browsers that don't support
  * unicode emoji rendering.
  */
-const EMOTICON_REGEXP_ARRAY: Array<Array<Object>> = [];
+const ASCII_EMOTICON_REGEXP_ARRAY: Array<Array<Object>> = [];
+
+/**
+ * An emoji regexp array to find and replace alias emoticons
+ * (such as :smiley:) with the new Unicode representation, so that
+ * devices and browsers that support them can render these natively
+ * without a 3rd party component.
+ *
+ * NOTE: this is currently only used on mobile, but it can be used
+ * on web too once we drop support for browsers that don't support
+ * unicode emoji rendering.
+ */
+const EMOJI_REGEXP_ARRAY: Array<Array<Object>> = [];
 
 (function() {
     for (const [ key, value ] of Object.entries(aliases)) {
-        let escapedValues;
-        const asciiEmojies = emojiAsciiAliases[key];
-
-        // Adding ascii emoticons
-        if (asciiEmojies) {
-            escapedValues = asciiEmojies.map(v => escapeRegexp(v));
-        } else {
-            escapedValues = [];
-        }
 
-        // Adding slack-type emoji format
-        escapedValues.push(escapeRegexp(`:${key}:`));
+        // Add ASCII emoticons
+        const asciiEmoticons = emojiAsciiAliases[key];
 
-        const regexp = `\\B(${escapedValues.join('|')})\\B`;
+        if (asciiEmoticons) {
+            const asciiEscapedValues = asciiEmoticons.map(v => escapeRegexp(v));
 
-        EMOTICON_REGEXP_ARRAY.push([ new RegExp(regexp, 'g'), value ]);
+            const asciiRegexp = `(${asciiEscapedValues.join('|')})`;
+
+            // Escape urls
+            const formattedAsciiRegexp = key === 'confused'
+                ? `(?=(${asciiRegexp}))(:(?!//).)`
+                : asciiRegexp;
+
+            ASCII_EMOTICON_REGEXP_ARRAY.push([ new RegExp(formattedAsciiRegexp, 'g'), value ]);
+        }
+
+        // Add slack-type emojis
+        const emojiRegexp = `\\B(${escapeRegexp(`:${key}:`)})\\B`;
+
+        EMOJI_REGEXP_ARRAY.push([ new RegExp(emojiRegexp, 'g'), value ]);
```

Comment:
would it make sense to name this `SLACK_EMOJI_REGEXP_ARRAY` and the other `ASCII_EMOJI_REGEXP_ARRAY` for consistency and more clarity?